botocore/httpchecksum.py

```diff
@@ -432,6 +435,16 @@ def _apply_request_trailer_checksum(request):
     )
 
 
+def _register_checksum_algorithm_feature_id(algorithm):
+    checksum_algorithm_name = algorithm["algorithm"]
```

Comment:
We should just make this `checksum_algorithm_name = algorithm["algorithm"].upper()` instead of having to do it each time we refer to it down below.